src/ui/public/visualize/visualize_legend.js

```diff
@@ -66,6 +66,18 @@ uiModules.get('kibana')
         return filters.length;
       };
 
+      $scope.showLabelCount = legendData => {
+        if (!$scope.vis.params.addLegendCount) return false;
+
+        const isBarChart = $scope.vis.type.name === 'histogram';
+        const isPieChart = $scope.vis.type.name === 'pie';
+        return  (isBarChart || isPieChart) && $scope.getLabelCount(legendData);
+      };
+
+      $scope.getLabelCount = legendData => {
```

Comment:
Consider renaming this to `getLegendCount` for consistency with `vis.params.addLegendCount`.